docs/tutorials/build_basic_model.ipynb

```diff
@@ -0,0 +1,118 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Build a basic model\n",
+    "\n",
+    "This _Build a basic model_ tutorial illustrates how to build a simple D-Flow FM model from scratch using the [HYDROLIB-core](https://github.com/Deltares/HYDROLIB-core). "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Import HYDROLIB-core functionalities and setup D-Flow FM model\n",
+    "from hydrolib.core.io.structure.models import FlowDirection, StructureModel, Weir\n",
+    "from hydrolib.core.io.mdu.models import FMModel\n",
+    "\n",
+    "fm = FMModel()\n",
+    "fm.filepath = \"test.mdu\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Add some structures, note this is invalid because it doesn't \n",
+    "have a branch or coordinates yet, but it will work for demo purposes"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Add weir to the model and assign structure to D-Flow FM model \n",
+    "struc = Weir(branchId='someBranch', chainage = 123.0, allowedflowdir=FlowDirection.none, crestlevel=0.0)\n",
+    "struc.comments.crestlevel = \"This is a comment\"\n",
+    "fm.geometry.structurefile = [StructureModel(structure=[struc])]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Note that the creation of a `Weir` and other models always require\n",
```

Comment:
Note that the creation of a `Weir` and other model objects requires several input arguments.
A `ValidationError` will be raised when the model is invalid or incomplete.
For instance, in the above example, if the `StructureModel` had been assigned directly to `structurefile` instead of as a list, that would have triggered a `ValidationError`.